frontend/src/pages/Groups/GroupPage/GroupPage.tsx

```diff
@@ -77,9 +79,15 @@ function GroupHub({ group }: GroupHubProps) {
   const { id } = useParams<{ id: string }>();
   const { goBack, push } = useHistory();
   const [leaveGroupMutation] = useMutation(LEAVE_GROUP);
+  const [joinGroupMutation] = useMutation(JOIN_GROUP);
+  const [checkMembershipMutation] = useMutation(CHECK_MEMBERSHIP);
   const { profile: user} = useProfileInfo();
+  const [loading, setLoading] = useState(true)
+  const [member, setMember] = useState(false)
+  const [leaving, setLeaving] = useState(false)
```

Comment:
loading and leaving have the same functionality. Maybe combine these into one variable. This would be more concise and deal with edge cases of user firing off mutations while leaving (which can currently happen).

Reply:
Just updated without member and leaving!